### PR TITLE
HIR-141. Add an endpoint to confirm a booking order cancellation

### PR DESCRIPTION
**Pull request by oleg1std**
Add an endpoint to confirm a booking order cancellation
Refactoring

### CODE REVIEW

File: HappyTravel.Hiroshima.DirectManager/Services/Bookings/BookingManagementService.cs [hunk 1]
```diff
@@ -31,19 +32,18 @@ public BookingManagementService(IManagerContextService managerContextService, Di
...
 
-            async Task<bool> AreAllAccommodationIdsBelongToServiceSupplier(ServiceSupplier serviceSupplier)
+            async Task<bool> AreAllAccommodationIdsBelongToSupplier(Manager manager)
             {
                 if (!bookingRequest.AccommodationIds.Any())
                     return true;
 
                 var existingIds = await _dbContext.Accommodations
-                    .Where(a => bookingRequest.AccommodationIds.Contains(a.Id) && a.ServiceSupplierId == serviceSupplier.Id)
+                    .Where(a => bookingRequest.AccommodationIds.Contains(a.Id) && a.ServiceSupplierId == manager.Id)
```
Comment (AlexeyShok) [inline HappyTravel.Hiroshima.DirectManager/Services/Bookings/BookingManagementService.cs:46]: Something is wrong here. ServiceSupplierId not equal ManagerId

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Hiroshima.DirectManager/Services/Bookings/BookingManagementService.cs [hunk 2]
```diff
@@ -72,21 +72,71 @@ async Task<List<BookingOrder>> GetBookings()
...
+                .Ensure(IsValidStatus, "Failed to confirm the booking order" )
+                .Bind(_ => _bookingService.Cancel(bookingId))
+                .Bind(bookingOrder => _bookingWebhookService.Send(bookingOrder.ReferenceCode, BookingStatuses.Cancelled));
+
+
+            bool IsValidStatus(BookingOrder bookingOrder) => bookingOrder.Status == BookingStatuses.WaitingForCancellation;
         }
+
         
+        public Task<Result> TryCancel(Guid bookingId)
```
Comment (mingazhev) [inline HappyTravel.Hiroshima.DirectManager/Services/Bookings/BookingManagementService.cs:102]: Maybe just "Cancel"?

**Review by mingazhev — APPROVED** [review]
With comments

File: HappyTravel.Hiroshima.DirectContracts/Services/BookingService.cs [hunk 3]
```diff
@@ -136,22 +136,34 @@ async Task<Result<Common.Models.Bookings.BookingOrder>> AddRoomOccupancy((Common
         }
 
 
-        public async Task<Result> Cancel(string referenceCode)
+        public Task<Result<Common.Models.Bookings.BookingOrder>> Confirm(Guid bookingId)
         {
-            return await Get(referenceCode)
-                .Ensure(BookingCanBeCancelled, $"The booking order '{referenceCode}' has already been cancelled")
+            return Get(bookingId)
+                .Ensure(IsBookingStatusValid, $"Failed to confirm the booking order with id '{bookingId}'")
```
[the remark this answers is not in the document]
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectContracts/Services/BookingService.cs:142]: Agree. The message has been changed

**Pull request merged**